src/machine/machine_rp2040-rtc.go

```diff
@@ -0,0 +1,256 @@
+//go:build rp2040
+
+// Implementation based on code located here:
+// https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/hardware_rtc/rtc.c
+
+package machine
+
+import (
+	"device/rp"
+	"errors"
+	"runtime/interrupt"
+	"runtime/volatile"
+	"unsafe"
+)
+
+type rtcType struct {
+	clkDivM1  volatile.Register32
+	setup0    volatile.Register32
+	setup1    volatile.Register32
+	ctrl      volatile.Register32
+	irqSetup0 volatile.Register32
+	irqSetup1 volatile.Register32
+	rtc1      volatile.Register32
+	rtc0      volatile.Register32
+	intR      volatile.Register32
+	intE      volatile.Register32
+	intF      volatile.Register32
+	intS      volatile.Register32
+}
```

Comment:
Why this new type definition? What's wrong with `rp.RTC_Type`?

Reply:
Nothing, it's just me not knowing it exists :D
Will fix.